### PR TITLE
ddl: fix creating partition table with unique prefix index

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

```
create table t1 (a varchar(20), b binary, unique index (a(5))) partition by range columns (a) (
    partition p0 values less than ('aaaaa'),
    partition p1 values less than ('bbbbb'),
    partition p2 values less than ('ccccc')
);
```

MySQL: A UNIQUE INDEX must include all columns in the table's partitioning function
TiDB: no error

Problem Summary:

Unique prefix index can't be used on the partition table.

### What is changed and how it works?

Proposal: [xxx](url) 

What's Changed:

How it Works:

Add check

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

### Release note 

- Fix the check of create and alter table, unique prefix index must not be used on the partition table

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @crazycs520 @zimulala @XuHuaiyu

File: ddl/partition.go [hunk 1]
```diff
@@ -761,6 +763,15 @@ func checkPartitioningKeysConstraints(sctx sessionctx.Context, s *ast.CreateTabl
 	return nil
 }
 
+func hasPrefixIndexColumn(idxCols []*model.IndexColumn) bool {
+	for _, col := range idxCols {
+		if col.Length > 0 {
```
Comment (crazycs520) [inline ddl/partition.go:768]: Should check the table name too?

Below sql will execute sucessfull in Mysql but faild in This PR:
```sql
create table part12 (a varchar(20), b varchar(10), unique index (a(5),b)) partition by range columns (b) (
			partition p0 values less than ('aaaaa'),
			partition p1 values less than ('bbbbb'),
			partition p2 values less than ('ccccc'))
```

Comment (tiancaiamao) [reply ddl/partition.go]: Fixed

**Pull request merged**